api/models.py

```diff
@@ -68,7 +76,7 @@ class Revision(BaseModel):
     describe: Optional[str] = None
 
 
-class Node(ModelId):
+class Node(DatabaseModel):
```

Comment:
Looking at the documentation:
https://docs.python.org/3/library/abc.html

It seems that `.register()` should be used for concrete sub-classes.

I was originally thinking of using "old-style" abstract classes by just not defining an implementation:
```python
class DatabaseModel(ModelId):

    def update(self):
        pass
```
Then models can inherit from it without using the `abc` module, and we can just call `obj.update()` on any of them.

@mgalka Do you think using `abc` is useful in this case?

Reply:
What do you suggest on this @mgalka ?

Reply:
> Looking at the documentation: https://docs.python.org/3/library/abc.html
> 
> It seems that `.register()` should be used for concrete sub-classes.
> 
> I was originally thinking of using "old-style" abstract classes by just not defining an implementation:
> 
> ```python
> class DatabaseModel(ModelId):
> 
>     def update(self):
>         pass
> ```
> 
> Then models can inherit from it without using the `abc` module, and we can just call `obj.update()` on any of them.
> 
> @mgalka Do you think using `abc` is useful in this case?

@gctucker The documentation https://docs.python.org/3/library/abc.html states that we will need to use `register()` method in case of virtual subclasses. On the contrary `Node` model explicitly inherits `DatabaseModel`. Hence, we do not need the method.

Reply:
If we go with the old-style implementation, we will be able to initiate DatabaseModel class. It won't be a pure abstract class then. It will be an inheritance instead. @mgalka @gctucker

Reply:
As far as I'm concerned, I'd keep the `ABC` mixin in the `DatabaseModel`. 

According to what I've read in the the previous comments there is an idea of having `Database.update()` method that'll call `update()` methods from models. Having `DatabaseModel` declared as an ABC will ensure that the `update()` method is actually implemented. Attempts to instantiate objects with unimplemented abstract methods can also be easily detected by static code checkers like `pylint` or `mypy`.
Although, having the "fully fledged" abstract class instead of an "old-style" one may seem a bit of an overhead as it has just one method it brings value to the project. I'd keep it.

@gctucker I'm not aware of a need of using `.register()` method unless we want to register our class as a sub-class of some unrelated class. It's more of a concept when regular inheritance doesn't seem quite right or e.g. we want to use a built-in class as a virtual class for our.

@JenySadadia I'd rename method `.set_updated()` to `.update()`. It looks more appropriate as the former suggest that it's implementation will be limited setting the `updated` class attribute and as far as I understand the idea behind the `.update()` method is to perform all kinds of actions that we need when updating the object in the database.

Reply:
Yes, that makes sense. @mgalka 
Renamed the method.

Reply:
Yes, actually my example is not really an abstract class since it has a default implementation (which does nothing).  A typical old-style abstract method with no implementation would raise `NotImplementedError`, like in some other places in kernelci-core.